src/pages/index.js

```diff
@@ -0,0 +1,356 @@
+  //3. handle the response (.then and .catch)
+  //4. in the .then, toggle active class
+  api
+    .changeLike(id, !isLiked)
+    .then(() => {
+      evt.target.classList.toggle("card__like-btn_liked");
+    })
+    .catch(console.error);
+}
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameElement = cardElement.querySelector(".card__title");
+  const cardImageElement = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  //TODO - if the card is liked, set the active class to the card
+
+  cardNameElement.textContent = data.name;
+  cardImageElement.src = data.link;
+  cardImageElement.setAttribute("alt", data.name);
+
+  cardLikeBtn.addEventListener("click", (evt) =>
+    //cardLikeBtn.classList.toggle("card__like-btn_liked");
+    handleLike(evt, data._id)
+  );
+
+  cardDeleteBtn.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  cardImageElement.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalCaption.textContent = data.name;
+    previewModalImage.src = data.link;
+    previewModalImage.setAttribute("alt", data.name);
+  });
+
+  return cardElement;
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscape);
+  modal.addEventListener("mousedown", handleOverlay);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscape);
+  modal.removeEventListener("mousedown", handleOverlay);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  //submitBtn.textContent = "Saving...";
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, true, "Save", "Saving...");
+    });
+}
+
+//TODO - implement loadng text for all other form submissions
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  const submitBtn = evt.submitter;
+
+  api
+    .createCards(inputValues)
+    .then((newCard) => {
+      const cardEl = getCardElement(newCard);
+      cardsList.prepend(cardEl);
+
+      disableButton(cardSubmitBtn, settings);
+
+      closeModal(cardModal);
+      cardForm.reset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, true, "Save", "Saving...");
```

Comment:
The `setButtonText` function has `true` is for the loading effect, but you should stop it in `.finally`


```js
      setButtonText(submitBtn, false);


```